src/mon/MonClient.cc

```diff
@@ -1368,3 +1271,93 @@ void MonClient::register_config_callback(md_config_t::config_callback fn) {
 md_config_t::config_callback MonClient::get_config_callback() {
   return config_cb;
 }
+
+class monc_error_category : public ceph::converting_category {
+public:
+  monc_error_category(){}
+  const char* name() const noexcept override;
+  std::string message(int ev) const noexcept override;
+  boost::system::error_condition default_error_condition(int ev) const noexcept
+    override;
+  bool equivalent(int ev, const boost::system::error_condition& c) const
+    noexcept override;
+  using ceph::converting_category::equivalent;
+  int from_code(int ev) const noexcept override;
+};
+
+const char* monc_error_category::name() const noexcept {
+  return "monc";
+}
+
+std::string monc_error_category::message(int ev) const noexcept {
+  using namespace ::monc_errc;
+
+
+  switch (ev) {
+  case 0:
+    return "No error";
+
+  case shutting_down: // Command failed due to MonClient shutting down
+    return "Command failed due to MonClient shutting down";
+  case session_reset:
+    return "Monitor session was reset";
+  case rank_dne:
+    return "Requested monitor rank does not exist";
+  case mon_dne:
+    return "Requested monitor does not exist";
+  case timed_out:
+    return "Monitor operation timed out";
+  }
+
+  return "Unknown error";
```

Comment:
Simliar conflicting feelings. It feels slightly wrong, but what else to do?

Reply:
Well, the obvious other thing to do is abort. And really since we're using an enum that might not be bad as the compiler can flag missing cases, though this is what most actual code and provided examples do.